src/erlcloud_s3.erl

```diff
@@ -1014,24 +1014,48 @@ make_link(Expire_time, BucketName, Key, Config) ->
       true  -> lists:flatten([Config#aws_config.s3_scheme, Config#aws_config.s3_host, port_spec(Config), "/", BucketName, "/", Key])
   end.
 
--spec make_get_url(integer(), string(), string()) -> iolist().
-
-make_get_url(Expire_time, BucketName, Key) ->
-    make_get_url(Expire_time, BucketName, Key, default_config()).
+ -spec make_get_url(integer(), string(), string()) -> iolist().
 
+make_get_url(ExpireTime, BucketName, Key) ->
+    make_get_url(ExpireTime, BucketName, Key, default_config()).
+  
 -spec make_get_url(integer(), string(), string(), aws_config()) -> iolist().
 
-make_get_url(Expire_time, BucketName, Key, Config) ->
-    {Sig, Expires} = sign_get(Expire_time, BucketName, erlcloud_http:url_encode_loose(Key), Config),
-    SecurityTokenQS = case Config#aws_config.security_token of
-        undefined -> "";
-        SecurityToken -> "&x-amz-security-token=" ++ erlcloud_http:url_encode(SecurityToken)
-    end,
-    lists:flatten([get_object_url(BucketName, Key, Config),
-     "?AWSAccessKeyId=", erlcloud_http:url_encode(Config#aws_config.access_key_id),
-     "&Signature=", erlcloud_http:url_encode(Sig),
-     "&Expires=", Expires,
-     SecurityTokenQS]).
+make_get_url(ExpireTime, BucketName, Key, Config) ->
+    Host = Config#aws_config.s3_host,
+    Region = aws_region_from_host(Host),
+    Path = erlcloud_http:url_encode_loose("/" ++ BucketName ++ "/" ++ Key),
+    EndpointUrl  = lists:flatten([
+      Config#aws_config.s3_scheme,
+      Host, port_spec(Config), Path]),
+    Headers = [{"Host", Host}],
+    Service = "s3",
```

Comment:
I can not using erlcloud_aws:sign_v4 result for signing temporary url. We need erlcloud_aws function that just sign request but not do and return anysing else. I thnk sign_v4 - good name for this function but everything else not related to signature can be moved to other function to prevent copypaste. But sign_v4 - public function and we can not change its interface. I added sign_v4_request and can move there all related to signature.